src/olympia/reviews/tests/test_views.py

```diff
@@ -996,6 +996,137 @@ def test_list_addon_and_user(self):
         assert len(data['results']) == 1
         assert data['results'][0]['id'] == recent_review.pk
 
+    def test_list_addon_and_version(self):
+        self.user = user_factory()
+        old_version = self.addon.current_version
+        other_version = version_factory(addon=self.addon)
+        old_review = Review.objects.create(
+            addon=self.addon, body='old review', user=self.user,
+            version=old_version)
+        old_review.update(created=self.days_ago(42))
+        other_review_same_addon = Review.objects.create(
+            addon=self.addon, body='review from other user',
+            user=user_factory(), version=old_version)
+        # None of those extra reviews should show up.
+        recent_review = Review.objects.create(
+            addon=self.addon, body='recent review', user=self.user,
+            version=other_version)
+        review_deleted = Review.objects.create(
+            addon=self.addon, body='review deleted', user=self.user)
+        review_deleted.delete()
+        Review.objects.create(
+            addon=other_review_same_addon.addon, body='reply to other user',
+            reply_to=other_review_same_addon, user=self.user)
+        Review.objects.create(
+            addon=addon_factory(), body='review from other addon',
+            user=self.user)
+
```

Comment:
Damn we really need factories to simplify this process and make it more readable 😨